### PR TITLE
Party/Feature: Initial implementation

### PR DESCRIPTION
**Pull request by MaxtorCoder**
**Features**
- Group
  - Group Creation
  - Member adding
  - More soon

**Todo:**
- [x] Request Join (_In Progress_)
- [x] Kick Member
- [x] Leave Group 
- [x] Convert to Raid
- [ ] Save group and members in Database (_for disconnect/server crash_)
- [x] Player/Unit marking
- [ ] Display member on map
- [x] Promote group member

**Authors**
- @MaxtorCoder 
- @alex-anderson156
- @albeva - Some code from their branch
- @zachstarnes - Helped @albeva

### CODE REVIEW

File: Source/NexusForever.WorldServer/Game/Group/Group.cs [hunk 1]
```diff
@@ -163,41 +163,136 @@ public GroupInvite CreateInvite(GroupMember inviter, Player invitedPlayer, Group
...
+        }
 
-            if (!isExpired)
+        /// <summary>
+        /// Accepts the <see cref="GroupInvite"/> and Adds the Invited Player to the group.
+        /// </summary>
+        /// <param name="invite">The <see cref="GroupInvite"/> to accept.</param>
+        public void AcceptInvite(string inviteeName)
+        {
+            GroupInvite invite = invites.Values.Single(inv => inv.TargetPlayer.Name.Equals(inviteeName));
```
Comment (Therzok) [inline Source/NexusForever.WorldServer/Game/Group/Group.cs]: SingleOrDefault can return null, Single will throw.

Comment (Therzok) [reply Source/NexusForever.WorldServer/Game/Group/Group.cs]:
```suggestion
            GroupInvite invite = invites.Values.SingleOrDefault(inv => inv.TargetPlayer.Name.Equals(inviteeName));
```

Comment (alex-anderson156) [reply Source/NexusForever.WorldServer/Game/Group/Group.cs]: your right, good spot.

**Comment by kirmmin** [comment]
If you wouldn't mind me merging this into my latest branch, is it possible to have this PR rebased onto develop and have it's commit history squashed, please?

**Comment by MaxtorCoder** [comment]
> If you wouldn't mind me merging this into my latest branch, is it possible to have this PR rebased onto develop and have it's commit history squashed, please?

I'll see what I can do today. 😄

**Comment by MaxtorCoder** [comment]
@kirmmin Done :)

**Comment by Rawaho** [comment]
Rebased to game_rework branch.

**Pull request merged**